packages/react-components/react-toolbar/src/components/ToolbarRadio/useToolbarRadio.ts

```diff
@@ -0,0 +1,43 @@
+import * as React from 'react';
+import { useToggleButton_unstable } from '@fluentui/react-button';
+import { useToolbarContext_unstable } from '../Toolbar/ToolbarContext';
+import { ToolbarRadioProps, ToolbarRadioState } from './ToolbarRadio.types';
+
+/**
+ * Given user props, defines default props for the ToggleButton, calls useButtonState and useChecked, and returns
+ * processed state.
+ * @param props - User provided props to the ToggleButton component.
+ * @param ref - User provided ref to be passed to the ToggleButton component.
+ */
+export const useToolbarRadio_unstable = (
+  props: ToolbarRadioProps,
+  ref: React.Ref<HTMLButtonElement | HTMLAnchorElement>,
+): ToolbarRadioState => {
+  const handleRadio = useToolbarContext_unstable(ctx => ctx.handleRadio);
+  const checked = useToolbarContext_unstable(ctx => !!ctx.checkedValues[props.name]?.includes(props.value));
+  const size = useToolbarContext_unstable(ctx => ctx.size);
+
+  const { onClick: onClickOriginal } = props;
+  const toggleButtonState = useToggleButton_unstable({ size, checked, ...props }, ref);
+  const state: ToolbarRadioState = {
+    ...toggleButtonState,
+    name: props.name,
+    value: props.value,
+  };
+
+  const handleOnClick = (
```

Comment:
Will wait for @bsunderhus , if he confirms I will remove the entire `disabled` if.

Reply:
Yes, since `useToggleButton` uses `useButton` under the hood `useARIAButton` methods should handle `disabled`, `aria-disabled` and `disabledFocusable` logic! And since `ToggleButtonProps` extends `ButtonProps` the proper properties will be there also 😁.

Also, would be nice to add a `useEventCallback` hook for this click handler.